### ISSUE TITLE
FileType auto detection fails on remote files which hosted on Amazon AWS

### ISSUE THREAD

**Issue by tomasr78** [ConvertAPI/convertapi-node#3]
The method below returns 404 HTTP exception when remote file of AWS passed. Needs to double check if remote file type detection works.

convertapi.convert('pdf', { File: 'https://website/my_file.docx' });

**Comment by laurynas-baltsoft** [comment]
Strange, I have just tested and format auto detection with S3 URL works fine for me. Format is auto-detected from the extension which is usually available in the S3 URL.

Could you provide failing example (url and conversion options)?

**Comment by tomasr78** [comment]
The problem was with AWS file permission and not with library, the case could be closed.

**Comment by tomasr78** [comment]
Another question related to the issue, could you confirm that remote files with query parameters works fine 

convertapi.convert('pdf', { File: 'https://cdn.convertapi.com/cara/testfiles/document.docx?test=1' });

**Comment by priyankinfinnov** [comment]
+1

**Comment by tomasr78** [comment]
@laurynas-baltsoft Just want to let you know that we implemented file extension detection directly into file upload rest api endpoint. Examples below. If you use file type detection inside NodeJS I strongly suggest use upload response to get file type. It has complex file detection which not only detects file from url but also from content disposition or even media type.

Local file upload

```
curl -i -X POST \
   -H "Content-Type:multipart/form-data" \
   -F "File=@\"./test.txt\";type=text/plain;filename=\"test.txt\"" \
 'https://v2.convertapi.com/upload'
```

result

```
{
"FileId": "LSFIGHRTNQUZI",
"FileName": "test.txt",
"FileExt": "txt"
}

```

Remote file upload

```
curl -i -X POST \
   -H "Content-Type:multipart/form-data" \
 'https://v2.convertapi.com/upload-from-url?url=https%3A%2F%2Fcdn.convertapi.com%2Fcara%2Ftestfiles%2Fdocument.docx%3Ftest%3D1'
```

result

```
{
"FileId": "LSF8DJLTT0PHH",
"FileName": "document.docx",
"FileExt": "docx"
}
```

**Comment by laurynas-baltsoft** [comment]
Fixed and released v1.1.0 version.
